### PR TITLE
[59849] Allow equality comparison between ipv4 and ipv6

### PR DESCRIPTION
**Pull request by Beetelbrox**
Issue: #59849
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for comparing IPv4 and IPv6 types using the `=` operator.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

You can check if IPv4 & IPv6 values are equal like any other pair of values:
```
SELECT toIPv4('127.0.0.1') = toIPv6('::ffff:127.0.0.1')
```

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>CI Settings</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Run these jobs only (required builds will be added automatically):
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with aarch64
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Deny these jobs:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

</details>

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit cd395ef346059620ddb1fa4898f6d9c9f8f0bd29 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64292/cd395ef346059620ddb1fa4898f6d9c9f8f0bd29/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

File: src/Functions/FunctionsComparison.h [hunk 1]
```diff
@@ -1176,8 +1176,7 @@ class FunctionComparison : public IFunction
             /// You can compare the date, datetime, or datatime64 and an enumeration with a constant string.
             || ((left.isDate() || left.isDate32() || left.isDateTime() || left.isDateTime64()) && (right.isDate() || right.isDate32() || right.isDateTime() || right.isDateTime64()) && left.idx == right.idx) /// only date vs date, or datetime vs datetime
             || (left.isUUID() && right.isUUID())
-            || (left.isIPv4() && right.isIPv4())
-            || (left.isIPv6() && right.isIPv6())
+            || ((left.isIPv4() || left.isIPv6()) && (left.isIPv4() || left.isIPv6()))
```
Comment (Beetelbrox) [inline src/Functions/FunctionsComparison.h:1179]: I just realized it only says "left" in both sides here, pushing a fix

Comment (Beetelbrox) [reply src/Functions/FunctionsComparison.h]: Fixed

**Review by pufit — APPROVED** [review]

**Comment by Beetelbrox** [comment]
We had a couple of upgradechecks failing, might be related to #64308? I've rebased to the latest master to see if that helps, otherwise will investigate further

**Comment by pufit** [comment]
@Beetelbrox, don't worry - those tests are unrelated. Our CI is stuck somewhere for whatever reason. I'll try to give a push to it.

**Pull request merged**